From: Sven Roederer <devel-sven@geroedel.de>
Date: Tue, 20 Apr 2021 02:17:38 +0200
Subject: build: add NPROC_OVERRIDE option, to manually limit parallel CPU-usage

The NPROC variable is used by some tools to control the parallel usage of
the CPUs. By default all available CPUs are used.
The new option can be used to keep some CPUs free and the system more
responsitive. This value can also prevent out of memory errors on low RAM
and high CPU systems.

Currently the NPROC variable is useed for compressing the imagebuilder and
sdk.

Signed-off-by: Sven Roederer <devel-sven@geroedel.de>

diff --git a/target/imagebuilder/Makefile b/target/imagebuilder/Makefile
index 0580e41c5790d8989ec2daf38e89e046cd1c0862..c69dc1df375a6c463c291d2a8702f216dfa5c33f 100644
--- a/target/imagebuilder/Makefile
+++ b/target/imagebuilder/Makefile
@@ -107,7 +107,7 @@ endif
 	$(CP) $(TOPDIR)/staging_dir/host/lib/libfakeroot* $(PKG_BUILD_DIR)/staging_dir/host/lib
 	STRIP=$(STAGING_DIR_HOST)/bin/sstrip $(SCRIPT_DIR)/rstrip.sh $(PKG_BUILD_DIR)/staging_dir/host/bin/
 	(cd $(BUILD_DIR); \
-		tar -I '$(STAGING_DIR_HOST)/bin/xz -7e -T$(if $(filter 1,$(NPROC)),2,0)' -cf $@ $(IB_NAME) \
+		tar -I '$(STAGING_DIR_HOST)/bin/xz -7e -T$(if $(NPROC_OVERRIDE),$(NPROC_OVERRIDE),$(if $(filter 1,$(NPROC)),2,0))' -cf $@ $(IB_NAME) \
 		--mtime="$(shell date --date=@$(SOURCE_DATE_EPOCH))"; \
 	)
 
diff --git a/target/sdk/Makefile b/target/sdk/Makefile
index 0606621192c7508fc7a470eff0173dc62ff70ded..bad3c192ed0ac8d80fe2f93364d54359b96006ca 100644
--- a/target/sdk/Makefile
+++ b/target/sdk/Makefile
@@ -164,7 +164,7 @@ $(BIN_DIR)/$(SDK_NAME).tar.xz: clean
 	find $(SDK_BUILD_DIR) -name CVS | $(XARGS) rm -rf
 	-make -C $(SDK_BUILD_DIR)/scripts/config clean
 	(cd $(BUILD_DIR); \
-		tar -I '$(STAGING_DIR_HOST)/bin/xz -7e -T$(if $(filter 1,$(NPROC)),2,0)' -cf $@ $(SDK_NAME) \
+		tar -I '$(STAGING_DIR_HOST)/bin/xz -7e -T$(if $(NPROC_OVERRIDE),$(NPROC_OVERRIDE),$(if $(filter 1,$(NPROC)),2,0))' -cf $@ $(SDK_NAME) \
 		--mtime="$(shell date --date=@$(SOURCE_DATE_EPOCH))"; \
 	)
 
